doc/source/contributing/docstrings.md

```diff
@@ -0,0 +1,68 @@
+(docstrings)=
+# Docstring formatting and type hints
+
+Docstrings should follow the
+[numpy docstring guide](https://numpydoc.readthedocs.io/en/latest/format.html).
+Extra guidance can also be found in
+[pandas docstring guide](https://pandas.pydata.org/pandas-docs/stable/development/contributing_docstring.html).
+Please reasonably document any additions or changes to the codebase,
+when in doubt, add a docstring.
+
+The different formatting and aim between numpydoc style type description and
+[type hints](https://docs.python.org/3/library/typing.html)
+should be noted. numpydoc style targets docstrings and aims to be human
+readable whereas type hints target function definitions and `.pyi` files and
+aim to help third party tools such as type checkers or IDEs. ArviZ does not
+require functions to include type hints
+however contributions including them are welcome.
+
+## References
+Docstrings follow the same guide for adding references as the other docs.
+For adding refernces to external libraries functions and objects, see {ref}`reference_external_libs`. For referencing ArviZ objects, follow {ref}`reference_arviz_objects`.
+
+## See Also
+For adding the _See Also_ docstring section, you just need to add the function name. Sphinx will
+automatically add links to other ArviZ objects and functions listed in the _See Also_
+section.
+
+For example, let's add {func}`~arviz.hdi` and {func}`~arviz.plot_ppc` in the _See Also_ section.
+
+```
+    See Also
+    --------
+    hdi : Calculate highest density interval (HDI) of array for given probability.
+    plot_ppc : plot for posterior/prior predictive checks.
+```
+
+## Kwargs parameters
```

Comment:
kwargs and backend_kwargs are the same aand should not be two sections. Moreover, the same reasoning used for kwargs applies to any function that calls another one. Again, I would focus **much** much more in why should things be done in a specific way and not so much on how to do those things.